Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -373,6 +373,49 @@ def evalPow : PositivityExt where eval {u α} zα pα e := do
     | .nonzero pa => ofNonzero pa (← synthInstanceQ (_ : Q(Type u)))
     | .none => pure .none
 
+set_option linter.deprecated false in
+/-- The `positivity` extension which identifies expressions of the form `a ^ (b : ℤ)`,
+such that `positivity` successfully recognises both `a` and `b`. -/
+@[positivity (_ : α) ^ (_ : ℤ), Pow.pow _ (_ : ℤ)]
+def evalZpow : PositivityExt where eval {u α} zα pα e := do
+  let .app (.app _ (a : Q($α))) (b : Q(ℤ)) ← withReducible (whnf e) | throwError "not ^"
+  let result ← catchNone do
+    let .true := b.isAppOfArity ``OfNat.ofNat 3 | throwError "not a ^ n where n is a literal"
+    let some n := (b.getRevArg! 1).natLit? | throwError "not a ^ n where n is a literal"
+    guard (n % 2 = 0)
+    have m : Q(ℕ) := mkRawNatLit (n / 2)
+    haveI' : $b =Q bit0 $m := ⟨⟩
+    let _a ← synthInstanceQ q(LinearOrderedField $α)
+    haveI' : $e =Q $a ^ $b := ⟨⟩
+    assumeInstancesCommute
+    pure (by exact .nonnegative q(zpow_bit0_nonneg $a $m))
```

Comment:
Is it possible to use `Even.zpow_nonneg` here, instead of `zpow_bit0_nonneg`? Would that remove the need for the `linter.deprecated false`?

Reply:
Yes (see the current version), but this adds extra imports (namely `Mathlib.Data.Int.Parity`) to get `Int.even_iff`, and the way it's coded now the `unusedHavesSuffices` linter flags a line that doesn't seem spurious (at least I don't know how to remove it). Doesn't seem worth it unless we're actually planning to remove `bit0` entirely.

Reply:
Hm... I think I agree that we should try to avoid adding that new dependency.

In your original code, you can dodge `linter.deprecated` by doing this:
```diff
-    haveI' : $b =Q bit0 $m := ⟨⟩
+    haveI' : $b =Q $m + $m := ⟨⟩ -- b = bit0 m
```

That seems better to me than disabling the linter for the entire declaration.